Features/Improvements/Bugfixes

vikramraman opened the pull request:
- Added OpenTSDB support (MONIT-7390)
- Export GC/Memory stats into Wavefront
- Addressed OOM issues (MONIT-7505): We drop points beyond configured limits for now, will queue to disk and re-send once disk buffering support is implemented.
- Log proxy summary once a minute
- Other minor code improvements

vikramraman committed:
Log metrics summary periodically

vikramraman committed:
removed extraneous logging

vikramraman committed:
Added OpenTSDB support

vikramraman committed:
MONIT-7575: Export GC/memory metrics

vikramraman committed:
Added support for cross compiling

vikramraman committed:
MONIT-7505: OOM while running 100K pps load

vikramraman committed:
Additional timestamp validation

vikramraman committed:
Removed fixed TODO

zhizhu7 commented on Makefile:
```diff
@@ -21,6 +21,9 @@ deps:
 proxy:
 	go build -i -o $(PROXY) -ldflags "$(LDFLAGS)" ./proxy/proxy.go
 
+amd64:
```
Besides building for the default OS, can we also include building binary for other OS, like Windows, Linux, darwin?

vikramraman replied:
Yes, I'm currently working on a ticket to improve the packaging for the proxy. This will be addressed in that.

vikramraman committed:
Moved main file to cmd directory

vikramraman committed:
Added --version flag

vikramraman committed:
Build packaging

vikramraman committed:
Added config file support

vikramraman committed:
Added default config file/minor code fixes

vikramraman commented:
- Added a new package and package-all Makefile targets that support building .deb, .rpm linux packages and windows/darwin executables. (MONIT-7576)
- Added config file support for the go-proxy (which is used by the linux systemd wavefront-proxy service) (MONIT-7396)
- Added a --version flag that prints the current version and git tag/commit info for the proxy.

zhizhu7 reviewed (approved):
LGTM.

vikramraman commented:
Merging to move along the QA process since Zhi has done reviewing it.
